vumi/transports/smpp/processors/default.py

```diff
@@ -470,6 +470,10 @@ class SubmitShortMessageProcessorConfig(Config):
         "If `True`, messages longer than 140 bytes will be sent as a series "
         "of smaller messages with the user data headers. Default is `False`.",
         default=False, static=True)
+    multipart_sar_reference_rollover = ConfigInt(
+        "The maximum value to set for the reference number of a multi part "
+        "SMS. When this value is reached, the number will rollover to 0.",
+        default=0xFFFF, static=True)
```

Comment:
This is actually a bitmask rather than a maximum value.

Maybe make the default `0x10000` and document that the reference resets to zero when it reaches that value or something? Then we can go back to using `%` and it will behave as expected for all possible config values.

Alternatively, maybe this should be `multipart_sar_reference_octets` with `1` and `2` as possible values? Then we can check that and either `& 0xFF` or `% 0xFFFF` as appropriate.